### ISSUE TITLE
Animate reposition of elements on the page when adding or removing an item

### ISSUE THREAD

**Issue by gdoyle1** [patternfly/patternfly-design#985]
**High priority**

For example, when dismissing a hint pattern, making the removal smoother and moving everything else up. This type of transition can be used for similar patterns/components as well, such as:

- Inline alert (both appearing and dismissing)
- Dismissing a hint
- Dashboard cards
- Data list / table rows
- Expandable sections of page

[Here's the prototype](https://xd.adobe.com/view/d878fc21-cea7-476c-70dd-685af6a31e1d-ae49/) that the animation group came up with. It would be good to look more into where this can be applied!

**Comment by KKoukiou** [comment]
Hi, we are really looking forward to this feature. Glad to see it in high priority already, In cockpit very often we feel like having a highlight / animatoin for new items in tables, lists etc will help the users follow better the activity.

**Comment by garrett** [comment]
We're looking forward to animation on new items; it'd solve a usability finding we had back in 2018 (which still exists and would affect every app using PatternFly).

Here's a probably-relevant CSS technique for animating new items that was recently covered:
https://css-tricks.com/a-handy-little-system-for-animated-entrances-in-css/

The important CSS is `animation-fill-mode: backwards;`

According to https://developer.mozilla.org/en-US/docs/Web/CSS/animation-fill-mode:

> The animation will apply the values defined in the first relevant keyframe as soon as it is applied to the target, and retain this during the animation-delay period. The first relevant keyframe depends on the value of animation-direction

In other words, it prevents the item from showing up before the animation starts.

**Comment by mcarrano** [comment]
Covered here for inclusion with PatternFly 5: https://docs.google.com/spreadsheets/d/1jjm2nXvqLsObuhSZ-cbII7d22WQ98W5r1ycC9BwJCIk/edit?usp=sharing

**Comment by andrew-ronaldson** [comment]
This is being picked up in another v6 issue

**Comment by garrett** [comment]
> This is being picked up in another v6 issue

Which issue? Can you provide a link?
